ts/Stock/Indicators/RSI/RSIIndicator.ts

```diff
@@ -106,20 +107,28 @@ class RSIIndicator extends SMAIndicator {
             change: number,
             avgGain: number,
             avgLoss: number,
-            i: number;
+            i: number,
+            values: Array<number>;
 
         // RSI requires close value
-        if (
-            (xVal.length < period) || !isArray(yVal[0]) ||
-            yVal[0].length !== 4
-        ) {
+        if ((xVal.length < period)) {
             return;
         }
+        if (!isArray(yVal[0])) {
+            values = yVal as Array<number>;
+        } else if (yVal[0].length < 4) {
```

Comment:
Why `yVal[0].length < 4`? If this is a preparation for cases like `HLC` series, could you add a comment explanation?

Reply:
well, in fact as you suggested earlier if I can calculate the index in a way: `Math.min(params.index, yVal[0].length - 1)` it 
can ensure, that we don't reach outside of the bounds of the array, so this `else if` is redundant.